### DISCUSSION TITLE
Include item link in Header

### DISCUSSION THREAD

**Discussion by macrospecter** [windingwind/zotero-better-notes#557] [Note Templates]
I'm hoping to include in the YAML data in the header a link back to the Zotero item. I don't care if it goes back to the note, the Zotero entry, or the PDF in the PDF reader--just some way to get back to the Zotero note from within the exported note. 

**Here is my current ExportMDFileHeaderV2 template, without the link:** 

`${await new Promise(async (r) => {
    let header = {};
    header.title = noteItem.parentItem
      ? noteItem.parentItem.getField("title")
      : "";
header.authors = noteItem.parentItem
      ? noteItem.parentItem.getCreators().map((au) => au.firstName + " " + au.lastName).join("; ")
      :"";
header.tags = noteItem.parentItem.getTags().map((_t) => _t.tag);
    r(JSON.stringify(header));
  })}`

**Ideally, the final note would look like this, with the link:**

title: x
authors: y
tags: z
link: a
[all required fields-note id, library id, etc.]

**I am aware of the following template for getting the PDF link**, and I can make it work for a note template, but I cannot seem to make it work within the ExportMDFileHeader template, despite a lot of trial and error and searching through the discussions here. 

`${{
  async function getPDFLink(item) {
    const att = await item.getBestAttachment();
    if (!att || !att.isPDFAttachment()) {
      return "";
    }
    key = att.key;
    if (att.libraryID === 1) {
      return `zotero://open-pdf/library/items/${key}`;
    } else {
      groupID = Zotero.Libraries.get(att.libraryID).id;
      return `zotero://open-pdf/groups/${groupID}/items/${key}`;
    }
  }
  sharedObj.getPDFLink = getPDFLink;
  return await getPDFLink(topItem);
}}$`

Would greatly appreciate anyone's help in how to implement this!

**Comment by macrospecter** [comment]
Thank you so much, but I think this template just puts the link in the body of the note. I’m trying to put it in the yaml header that only shows in the note exported to Obsidian, not in Zotero. Do you by chance know how to do that? It involves editing the Header default template, and that file seems to accept a different coding language only.

**Reply by nilleo** [reply]
Sorry, I also don't know how to solve your problem. Maybe you can submit an issue or change the category of this discussion to Q&A? I replied without understanding your question earlier, but I realized it and deleted it. Sorry for the disturbance.

**Comment by windingwind** [comment]
```yaml
# This template is specifically for importing/sharing, using better 
# notes 'import from clipboard': copy the content and
# goto Zotero menu bar, click Edit->New Template from Clipboard.  
# Do not copy-paste this to better notes template editor directly.
name: "[ExportMDFileHeaderV2]"
content: |-
  ${await new Promise(async (r) => {
      let header = {};
      header.tags = noteItem.getTags().map((_t) => _t.tag);
      header.parent = noteItem.parentItem
        ? noteItem.parentItem.getField("title")
        : "";
      header.collections = (
        await Zotero.Collections.getCollectionsContainingItems([
          (noteItem.parentItem || noteItem).id,
        ])
      ).map((c) => c.name);
      header.noteLink = Zotero.BetterNotes.api.convert.note2link(noteItem);
      r(JSON.stringify(header));
    })}
```

**Reply by windingwind** [reply]
This is the minimal implementation based on the default template.

**Reply by macrospecter** [reply]
Thank you so much. This template produces a link that seems unique to the Zotero note, but it's not in a clickable form. I'm not quite sure how to use it to access the note. What I'm hoping for is a link that immediately opens the note in Zotero (I do have a template that allows me to put such a link in the body of the note, I'm just hoping to move it to the header). Is that possible?

(For others' reference, the template above from windingwind may need to be copied using the copy icon. When I selected and used copy/paste, one of the symbols was transformed into a string of letters and the symbol &, which led to an error within Zotero.)

**Reply by windingwind** [reply]
I'm not sure how your markdown viewer will display the links in the YAML header. YAML is a pure text format and isn't designed to provide clickable links unless your markdown editor supports it.

You can copy the link and paste it into your browser to navigate to the note in Zotero.

**Reply by macrospecter** [reply]
Ah now I see (I know very little about YAML). Thank you very much anyway--I'll see if what I'm trying to do is possible from within my markdown editor. 

In the event that that isn't possible, I do have a separate question about the YAML header, because I may have to repurpose mine. I'm opening as a new template discussion issue [here](https://github.com/windingwind/zotero-better-notes/discussions/592#discussion-5451418), in case anyone has the same question.
